src/msd_lite.c

```diff
@@ -643,6 +702,64 @@ msd_http_srv_on_req_rcv_cb(http_srv_cli_p cli, void *udata __unused,
 		return (HTTP_SRV_CB_NONE);
 	}
 
+	/* JSON Statistic request. */
+	if (HTTP_REQ_METHOD_GET == req->line.method_code &&
+	    0 == mem_cmpin_cstr("/api/stats", req->line.abs_path, req->line.abs_path_size)) {
+		
+		/* Auth Check */
+		if (g_data.admin_auth_basic) {
+			const uint8_t *val;
+			size_t val_len;
+			if (0 != http_hdr_val_get(req->hdr, req->hdr_size, (const uint8_t*)"Authorization", 13, &val, &val_len) ||
+			    val_len < 6 || 
+			    0 != memcmp(val, "Basic ", 6) ||
+			    val_len - 6 != g_data.admin_auth_basic_size ||
+			    0 != memcmp(val + 6, g_data.admin_auth_basic, g_data.admin_auth_basic_size)) {
+				resp->status_code = 401;
+				resp->hdrs_count = 1;
+				resp->hdrs[0].iov_base = MK_RW_PTR(auth_fail_hdrs);
+				resp->hdrs[0].iov_len = 46;
+				return (HTTP_SRV_CB_CONTINUE);
+			}
+		}
+
+		error = gen_hub_stat_json_send_async(g_data.shbskt, cli);
+		if (0 != error) {
+			resp->status_code = 500;
+			return (HTTP_SRV_CB_CONTINUE);
+		}
+		/* Will send reply later... */
+		return (HTTP_SRV_CB_NONE);
+	}
+
+	/* Admin page request. */
+	if (HTTP_REQ_METHOD_GET == req->line.method_code &&
+	    0 == mem_cmpin_cstr("/admin", req->line.abs_path, req->line.abs_path_size)) {
+		
+		/* Auth Check */
+		if (g_data.admin_auth_basic) {
```

Comment:
Please make auth check as function, since it is big enough to copy-paste it.
Also there is no sense in this since you do not require auth for ```/stat``` anb ```/hubstat``` URLs.